packages/proxy/src/handler.ts

```diff
@@ -65,41 +66,91 @@ function isRedirect(
 }
 
 export async function handler(event: CloudFrontRequestEvent) {
-  const { request } = event.Records[0].cf;
-  const configEndpoint = request.origin!.s3!.customHeaders[
-    'x-env-config-endpoint'
-  ][0].value;
-  const apiEndpoint = request.origin!.s3!.customHeaders['x-env-api-endpoint'][0]
-    .value;
+  const request = event.Records[0]?.cf.request;
+
+  if (request === undefined) {
+    console.error('Could not find any records in CF event.');
+    return request;
+  }
+
+  const customHeaders = request.origin?.s3?.customHeaders;
+
+  // Custom headers should exist, because we assign them to the CF
+  // distribution in the TF code.
+  if (customHeaders === undefined) {
+    console.error('Could not find custom headers in request.');
+    return request;
+  }
+
+  const domainName = customHeaders['x-env-domain-name']?.[0]?.value;
+  const configEndpoint = customHeaders['x-env-config-endpoint']?.[0]?.value;
+  let apiEndpoint = customHeaders['x-env-api-endpoint']?.[0]?.value;
+
+  if (configEndpoint === undefined || apiEndpoint === undefined) {
+    console.error('Could not find required endpoints in custom headers.');
+    return request;
+  }
+
   let headers: Record<string, string> = {};
+  let deploymentIdentifier = undefined;
+
+  // We need to re-fetch the proxy config for every request, because it could be
```

Comment:
So you would keep the configs in the cache for the whole hour that the lambda exists or you would associate the configs with a timestamp and clean them after a minute?

Reply:
The current behavior is that we keep the config in the cache until the Lambda dies which is not the best solution since the propagation of route changes could take very long this way.
Would rather go with the timestamp here so that we can control how long the config should be cached.

Reply:
I would suggest adding this in a separate PR. What do you think?

Reply:
Yep, it's out of scope for this PR 👍